lib/http/headers.rb

```diff
@@ -49,16 +53,30 @@ def delete(name)
     # @param [Array<#to_s>, #to_s] value header value(s) to be appended
     # @return [void]
     def add(name, value)
-      name = normalize_header name.to_s
-      Array(value).each { |v| @pile << [name, validate_value(v)] }
+      lookup_name = normalize_header(name.to_s)
+      wire_name = case name
+                  when String
+                    name
+                  when Symbol
+                    lookup_name
+                  else
+                    raise HTTP::HeaderError, "HTTP header must be a String or Symbol: #{name.inspect}"
```

Comment:
I think it should raise only if key is no ot responding to to_s and not a Symbol. Something  like:

``` ruby
if name.is_a? Symbol
  # ...
elsif name.respond_to? :to_s
  # ...
else
  raise
end
```

Reply:
I don't see how that is helpful. Almost everything in ruby responds to `to_s` (unless it derives from `BasicObject` to explicitly opt out of default behavior).

If you want to allow more than just `String` (not sure what the use case is), we could support anything that responds to `to_str`, which is used to support implicit conversion to strings.

Reply:
Ok. I agree. Then we need to update api doc to reflect this change.

Reply:
Do you want to want to keep my current strict implementation (only allow String or Symbol), or do you want to use the more relaxed "anything that supports an implicit conversion to string (ie. responds to `#to_str`)"?

I think being strict makes sense here, especially since we have different behavior based on the data type.

Reply:
I'm fine with your way. Strict Symbol or String.